### PR TITLE
Multiple improvements for uniforms-material package

### LINKED ISSUE #501: Warning Invalid prop `margin` of value `normal` supplied to `Input`
Getting browser warning caused by new changes in `uniforms-material 1.30.0` release as invalid props are being passed to the Material-UI [Input component](https://material-ui.com/api/input/). I verified that this warning is not happening with 1.29.0.

Removing `margin: 'normal'` from the `defaultProps` in `SelectField.js` resolves the warning. I'm using `Material-UI 3.8.2`.

```
modules.js?hash=38923953ebd0ef1fc7cb32a8b61a55d9e65ec9d4:44223 Warning: Failed prop type: Invalid prop `margin` of value `normal` supplied to `Input`, expected one of ["dense","none"].
    in Input (created by WithStyles(Input))
    in WithStyles(Input)
    in Select (created by Context.Consumer)
    in WithFormControlContext(Select) (created by WithStyles(WithFormControlContext(Select)))
    in WithStyles(WithFormControlContext(Select)) (created by Select)
    in div (created by FormControl)
    in FormControl (created by WithStyles(FormControl))
    in WithStyles(FormControl) (created by Select)
    in Select (created by SelectField)
    in SelectField (created by ContactsFilters)
    in form (created by AutoValidatedQuickMaterialForm)
    in AutoValidatedQuickMaterialForm (created by ContactsFilters)
    in ContactsFilters (created by WithStyles(ContactsFilters))
    in WithStyles(ContactsFilters) (created by Composer(WithStyles(ContactsFilters)))
    in Composer(WithStyles(ContactsFilters)) (created by Contacts)
    in h6 (created by Typography)
    in Typography (created by WithStyles(Typography))
    in WithStyles(Typography) (created by Layout)
    in div (created by Toolbar)
    in Toolbar (created by WithStyles(Toolbar))
    in WithStyles(Toolbar) (created by Layout)
    in header (created by Paper)
    in Paper (created by WithStyles(Paper))
    in WithStyles(Paper) (created by AppBar)
    in AppBar (created by WithStyles(AppBar))
    in WithStyles(AppBar) (created by Layout)
    in div (created by Layout)
    in Layout (created by WithWidth)
    in EventListener (created by WithWidth)
    in WithWidth (created by WithStyles(WithWidth))
    in WithStyles(WithWidth) (created by RouterContext)
    in RouterContext (created by Router)
    in Router
    in MuiThemeProviderOld
    in Provider
```

### PR DESCRIPTION
**Pull request by janowsiany**
- [x] Fixes #501 
- [x] Removes other warnings about unrecognised props
- [x] DateField using TextField from `material-ui`. Fixes `outlined` DateField layout
- [x] SelectField  using TextField from `material-ui`. Fixes `outlined` SelectField layout

### CODE REVIEW

**Comment by radekmie** [comment]
Can you somehow incorporate or comment on #502? I'd either close it or limit this one.

File: API.md [hunk 1]
```diff
@@ -326,11 +326,6 @@ import DateField from 'uniforms-unstyled/DateField'; // Choose your theme packag
     //   use this prop instead.
     inputRef={ref => {}}
 
-    // Props for the InputLabel
-    // Available in:
-    //   material-ui
-    labelProps={{ shrink: true, disableAnimation: true }}
```
Comment (radekmie) [inline API.md]: I'm not sure: can we keep it not to break the compatibility?

File: packages/uniforms-material/src/DateField.js [hunk 2]
```diff
@@ -1,11 +1,8 @@
-import Input from '@material-ui/core/Input';
-import InputLabel from '@material-ui/core/InputLabel';
+import TextField from '@material-ui/core/TextField';
```
Comment (radekmie) [inline packages/uniforms-material/src/DateField.js]: Ordering.

File: packages/uniforms-material/src/SelectField.js [hunk 3]
```diff
@@ -3,12 +3,11 @@ import FormControlLabel from '@material-ui/core/FormControlLabel';
 import FormGroup from '@material-ui/core/FormGroup';
 import FormHelperText from '@material-ui/core/FormHelperText';
 import FormLabel from '@material-ui/core/FormLabel';
-import InputLabel from '@material-ui/core/InputLabel';
 import MenuItem from '@material-ui/core/MenuItem';
 import Radio from '@material-ui/core/Radio';
 import RadioGroup from '@material-ui/core/RadioGroup';
 import React from 'react';
-import SelectMaterial from '@material-ui/core/Select';
+import TextField from '@material-ui/core/TextField';
```
Comment (radekmie) [inline packages/uniforms-material/src/SelectField.js]: Ordering.

File: packages/uniforms-material/src/SelectField.js [hunk 4]
```diff
@@ -27,40 +26,52 @@ const xor = (item, array) => {
 const renderSelect = ({
   allowedValues,
   disabled,
+  error,
+  errorMessage,
   fieldType,
+  fullWidth,
+  helperText,
   id,
+  InputLabelProps,
```
Comment (radekmie) [inline packages/uniforms-material/src/SelectField.js]: I'd rather keep it compatible... Maybe merge the old and the new one?

Also, ordering.

File: packages/uniforms-material/src/SelectField.js [hunk 5]
```diff
@@ -27,40 +26,52 @@ const xor = (item, array) => {
...
-      <InputLabel htmlFor={name} shrink={!!placeholder || !!value} {...labelProps}>
-        {label}
-      </InputLabel>
-    ),
-    <SelectMaterial
-      displayEmpty={!!placeholder}
-      inputProps={{name, id, ...inputProps}}
-      multiple={fieldType === Array || undefined}
-      native={native}
+  return (
```
Comment (radekmie) [inline packages/uniforms-material/src/SelectField.js]: What about `wrapField`?

File: packages/uniforms-material/src/DateField.js [hunk 6]
```diff
@@ -17,31 +14,38 @@ const dateParse = (timestamp, onChange) => {
...
+  inputRef,
+  label,
+  name,
+  onChange,
+  placeholder,
+  showInlineError,
+  value,
+  ...props
+}) => (
+  <TextField
```
Comment (radekmie) [inline packages/uniforms-material/src/DateField.js]: What about `wrapField`?

File: packages/uniforms-material/src/SelectField.js (hunk 5 above)
Comment (janowsiany) [reply packages/uniforms-material/src/SelectField.js]: TextField does not require `wrapField`.

File: packages/uniforms-material/src/DateField.js (hunk 6 above)
Comment (janowsiany) [reply packages/uniforms-material/src/DateField.js]: TextField does not require `wrapField`.

**Comment by Floriferous** [comment]
If I simply want all material-ui components to use the `outlined` variant, how can I do that in the simplest way?

**Comment by janowsiany** [comment]
👋I think [default props](https://material-ui.com/customization/globals/#default-props) is the way, let us know if that works for you!

**Pull request merged**